src/algebra/field/basic.lean

```diff
@@ -6,17 +6,18 @@ Authors: Robert Lewis, Leonardo de Moura, Johannes Hölzl, Mario Carneiro
 import algebra.hom.ring
 
 /-!
-# Fields and division rings
+# Division (semi)rings and (semi)fields
 
 This file introduces fields and division rings (also known as skewfields) and proves some basic
 statements about them. For a more extensive theory of fields, see the `field_theory` folder.
 
 ## Main definitions
 
-* `division_ring`: introduces the notion of a division ring as a `ring` such that `0 ≠ 1` and
-  `a * a⁻¹ = 1` for `a ≠ 0`
-* `field`: a division ring which is also a commutative ring.
-* `is_field`: a predicate on a ring that it is a field, i.e. that the multiplication is commutative,
+* `division_semiring`: Nontrivial semiring with multiplicative inverses for nonzero elements.
+* `division_ring`: : Nontrivial ring with multiplicative inverses for nonzero elements.
+* `semifield`: Commutative division semiring.
+* `field`: Commutative division ring.
+* `is_field`: Predicate on a ring that it is a field, i.e. that the multiplication is commutative,
```

Comment:
So this can also be used on a semiring `R` to claim that `R` is a semifield, right? Is it worth pointing that out explicitly?

Reply:
I'm not so sure. We didn't have `semifield`s. I think it's very cheap to include that generalization.